packages/vital_health/vital_health_ios/lib/vital_health_ios.dart

```diff
@@ -140,10 +140,177 @@ class VitalHealthIos extends VitalHealthPlatform {
     ]);
   }
 
+  @override
+  Future<ProcessedData> read(
+      HealthResource resource, DateTime startDate, DateTime endDate) async {
+    final result = await _channel.invokeMethod('read', [
+      resource.name,
+      startDate.millisecondsSinceEpoch,
+      endDate.millisecondsSinceEpoch
+    ]);
+
+    return _mapJsonToProcessedData(resource, jsonDecode(result));
```

Comment:
Note: #30 now propagates Swift errors by encoding them as `ErrorResult` and returning them via the channel.

But we haven't yet been handling/expecting `ErrorResult` on most of the channel consumer on the Dart side, except `ask`.

Reply:
Indeed, I will create a separate ticket for that. https://linear.app/try-vital/issue/VIT-2430/handle-errorresults-in-flutter